### DISCUSSION TITLE
Custom gcode settings for sequential printing?

### DISCUSSION THREAD

**Discussion by JstBecause** [supermerill/SuperSlicer#3140] [General]
So, when you throw custom gcodes on an object, it works great. But, it doesn't work if you use sequential printing. It shows the setting, but says it's impossible to use it. 

Which makes you need to go in and modify the gcode yourself. Which isn't horrible, however I think it should keep the settings.

Is it possible to allow us to use it in this manner?

For example. I have an object that needs different speed for the first 20 layers. So that's two speed changes per object. This is needed for the TPU prints I do.

**Comment by neophyl** [comment]
So just create a new printer profile for these jobs and add some custom gcode into the various custom gcode settings and use some layer based conditionals to control things.  Its quite easy to slow the whole printer down for lower areas and then speed it back up to normal above that.    

If you are using Marlin then M220 S80 would set the overall speed to 80% of normal.  Stick something like the following into the before layer section 
M220 S{if layer_z<2}80{else}100{endif} ; will change to 80% below 2mm

That makes the first 2 mm of each object print at 80% of normal speed and at 100% above that.  
Please be aware that the last M220 setting is persistent, so if you run a new print that doesn't set M220 then it will also print at whatever the last used.  Turning the printer off/on resets it though.  Its one reason that some of use add a M220 S100 into our start gcode block by default in case we forget :)
